Source/Android/jni/AchievementAdapter.cpp

```diff
@@ -55,4 +55,55 @@ Java_org_dolphinemu_dolphinemu_features_settings_model_AchievementModel_shutdown
+            env->SetObjectField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "badgeName", "Ljava/lang/String;"), env->NewStringUTF(achievement_list->buckets[ix].achievements[jx]->badge_name));
+            env->SetObjectField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "measuredProgress", "Ljava/lang/String;"), env->NewStringUTF(achievement_list->buckets[ix].achievements[jx]->measured_progress));
+            env->SetFloatField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "measuredPercent", "F"), achievement_list->buckets[ix].achievements[jx]->measured_percent);
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "id", "I"), achievement_list->buckets[ix].achievements[jx]->id);
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "points", "I"), achievement_list->buckets[ix].achievements[jx]->points);
+            // TODO: Convert to string? Or something.
+            //env->SetObjectField(bucket, env->GetFieldID(IDCache::GetAchievementClass(), "unlockTime", "Ljava/lang/String;"), env->NewStringUTF(achievement_list->buckets[ix].achievements[jx]->unlock_time));
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "state", "I"), achievement_list->buckets[ix].achievements[jx]->state);
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "category", "I"), achievement_list->buckets[ix].achievements[jx]->category);
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "bucket", "I"), achievement_list->buckets[ix].achievements[jx]->bucket);
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "unlocked", "I"), achievement_list->buckets[ix].achievements[jx]->unlocked);
+            env->SetFloatField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "rarity", "F"), achievement_list->buckets[ix].achievements[jx]->rarity);
+            env->SetFloatField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "rarityHardcore", "F"), achievement_list->buckets[ix].achievements[jx]->rarity_hardcore);
+            env->SetIntField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "type", "I"), achievement_list->buckets[ix].achievements[jx]->type);
+            env->SetObjectField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "badgeUrl", "Ljava/lang/String;"), env->NewStringUTF(achievement_list->buckets[ix].achievements[jx]->badge_url));
+            env->SetObjectField(achievement, env->GetFieldID(IDCache::GetAchievementClass(), "badgeLockedUrl", "Ljava/lang/String;"), env->NewStringUTF(achievement_list->buckets[ix].achievements[jx]->badge_locked_url));
```

Comment:
For performance reasons, I recommend getting the field IDs once in IDCache.cpp so that you don't have to get the field IDs all over again for each achievement. (Yes, I realize this is very repetitive to implement.)